app/controlplane/internal/service/group.go

```diff
@@ -531,6 +550,79 @@ func (g *GroupService) ListProjects(ctx context.Context, req *pb.GroupServiceLis
 	}, nil
 }
 
+// userHasPermissionToAddGroupMember checks if the user has permission to add members to a group
+func (g *GroupService) userHasPermissionToAddGroupMember(ctx context.Context, orgID string, groupIdentifier *pb.IdentityReference) error {
+	return g.userHasPermissionOnGroupMembershipsWithPolicy(ctx, orgID, groupIdentifier, authz.PolicyGroupAddMemberships)
+}
+
+// userHasPermissionToRemoveGroupMember checks if the user has permission to remove members from a group
+func (g *GroupService) userHasPermissionToRemoveGroupMember(ctx context.Context, orgID string, groupIdentifier *pb.IdentityReference) error {
+	return g.userHasPermissionOnGroupMembershipsWithPolicy(ctx, orgID, groupIdentifier, authz.PolicyGroupRemoveMemberships)
+}
+
+// userHasPermissionToListPendingGroupInvitations checks if the user has permission to list pending group invitations
+func (g *GroupService) userHasPermissionToListPendingGroupInvitations(ctx context.Context, orgID string, groupIdentifier *pb.IdentityReference) error {
+	return g.userHasPermissionOnGroupMembershipsWithPolicy(ctx, orgID, groupIdentifier, authz.PolicyGroupListPendingInvitations)
+}
+
+// userHasPermissionToListGroupMember checks if the user has permission to list group members
+func (g *GroupService) userHasPermissionToListGroupMember(ctx context.Context, orgID string, groupIdentifier *pb.IdentityReference) error {
```

Comment:
Moved all these methods from servier to `GroupService` and this is the new one that checks the policy for listing members in a group